### PR TITLE
[Bitcoin] bitseed validity

### PR DESCRIPTION
**Pull request by yubing744**
## Summary

Summary about this PR

- Closes #issue

### CODE REVIEW

File: frameworks/bitcoin-move/sources/light_client.move [hunk 1]
```diff
@@ -201,6 +201,12 @@ module bitcoin_move::light_client{
 
         // create new utxo
         handle_new_utxo(tx, &mut output_seals);
+
+        // handle bitseed
+        if(data_import_config::is_ord_mode(data_import_mode)) {
+            bitseed::process(tx);
```
Comment (yubing744) [inline frameworks/bitcoin-move/sources/light_client.move:207]: @jolestar I put the entry for parsing the bitseed protocol into light_client.move. Do you think it’s okay?

File: frameworks/bitcoin-move/sources/bitseed.move [hunk 2]
```diff
@@ -180,4 +213,45 @@ module bitcoin_move::bitseed {
...
+        };
+    }
+
+    public fun process_inscription(tx: &Transaction, inscription: &Inscription) {
+        let txid = types::tx_id(tx);
+        let index = ord::index(inscription);
+        let inscription_id = ord::new_inscription_id(txid, index);
+
+        if (is_bitseed(inscription)) {
+            // TODO parse inscription.meta and valid op
```
Comment (yubing744) [inline frameworks/bitcoin-move/sources/bitseed.move:253]: Since the meta of Inscription is cbor encoded, a decoding function is now needed.

**Comment by yubing744** [comment]
image

After rebasing the main branch, compiling frameworks reports an error with no useful error message

File: frameworks/bitcoin-move/sources/ord.move [hunk 3]
```diff
@@ -934,7 +951,22 @@ module bitcoin_move::ord {
         object::upsert_field(inscription_obj, METAPROTOCOL_VALIDITY, validity);
     }
 
-    // Borrow the metaprotocol validity for the given inscription_id.
+    /// Returns true if Inscription `object` contains metaprotocol validity
+    public fun exists_metaprotocol_validity(inscription_id: InscriptionID): bool{
+        let store_obj_id = object::named_object_id<InscriptionStore>();
+        let store_obj = object::borrow_mut_object_shared<InscriptionStore>(store_obj_id);
```
Comment (jolestar) [inline frameworks/bitcoin-move/sources/ord.move:957]: Do not need to `borrow_mut`.

File: frameworks/bitcoin-move/sources/ord.move [hunk 4]
```diff
@@ -934,7 +951,22 @@ module bitcoin_move::ord {
...
+
+        let inscription_object_id = derive_inscription_id(inscription_id);
+        let exists = object::contains_object_field<InscriptionStore, Inscription>(store_obj, inscription_object_id);
+        if (!exists) {
+            return false
+        };
+
+        let inscription_obj = object::borrow_mut_object_field<InscriptionStore, Inscription>(store_obj, inscription_object_id);
```
Comment (jolestar) [inline frameworks/bitcoin-move/sources/ord.move:965]: provide a function `borrow_inscription_option(InscriptionID):Option<Inscription>` ?

File: crates/testsuite/tests/images/bitseed.rs
Comment (jolestar) [inline crates/testsuite/tests/images/bitseed.rs:1]: We can migrate the `bitseed` command to `rooch bitseed` in the future.

**Review by jolestar — APPROVED** [review]

**Comment by jolestar** [comment]
> image
> 
> After rebasing the main branch, compiling frameworks reports an error with no useful error message
> 
> @jolestar Please help me look at this problem. Use the following command to trigger
> 
> ```
> bash ./scripts/bitcoin/test.sh -u
> ```

Because the ord.move and bitcoin.move contains break change.

**Comment by jolestar** [comment]
@steelgeek091 I change the verifier, allow `std::string::String` and `std::ascii::String` as `data_struct`.

**Review by steelgeek091 — APPROVED** [review]

**Comment by jolestar** [comment]
'bitseed/bitseed:0.1.4' needs to be pulled before testing.

I merge this PR first.

**Pull request merged**